src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceArr = sourceString.split(';');
+  const newArr = sourceArr.map((el) => el.trim());
+  const result = newArr.reduce((prev, el) => {
+    const [key, value] = el.split(':');
+
+    if (key && value) {
+      prev[key.trim()] = value.trim();
+    }
+
+    return prev;
+  }, {});
```

Comment:
[CODE PRACTICE] - Instead of using a loop inside `reduce`, consider using a filter to remove empty or invalid entries before the `reduce` function. This will make the code cleaner and separate the concerns of filtering and object construction.